### PR TITLE
feat: add RTC denoising to Pi05

### PR DESCRIPTION
**Pull request by eugene123tw**
# Pull Request

## Description

Add Real-Time Chunking (RTC) support to the Pi0.5 model and export pipeline. RTC applies guidance correction during the flow-matching denoising loop to improve temporal consistency of action predictions by blending the current prediction with the previous action chunk using adaptive prefix attention weights.

## Type of Change

- [x] ✨ `feat` - New feature

## Breaking Changes

None. The RTC functionality is opt-in via the `enable_rtc` flag during export and `rtc_prev_action_chunk` parameter during inference. Existing behavior is unchanged when RTC parameters are not provided.

---

## Examples

**Exporting with RTC enabled:**
```python
policy.export(output_dir="model_rtc", enable_rtc=True)
```

**Inference with RTC guidance (called internally by RTCActionChunking runner):**
```python
actions = model.sample_actions(
    images, img_masks, tokens, masks,
    rtc_max_guidance=10.0,
    rtc_execution_horizon=10,
    rtc_latency=8,
    rtc_prev_action_chunk=prev_chunk,
)
```

## Changes Made

- **mixin_model.py** — Added abstract `sample_input_rtc` property to `ExportableModelMixin` for RTC-specific export inputs.
- **mixin_policy.py** — Added `enable_rtc` export kwarg support; routes to `sample_input_rtc` when enabled. 
- **model.py** — Core RTC implementation:
  - `sample_input_rtc` property: generates sample inputs with RTC tensors (`prev_chunk_left_over`, `inference_delay`, `max_guidance_weight`, `execution_horizon`).
  - `_compute_prefix_weights()`: computes linear/exponential attention weights for blending previous and current chunks.
  - `_rtc_correct()`: applies adaptive guidance correction to flow-matching velocity, designed for OpenVINO traceability (without gradient).
  - Updated `sample_actions()` to accept RTC parameters and apply correction inside the denoising loop.

### CODE REVIEW

File: library/src/physicalai/policies/act/model.py [hunk 1]
```diff
@@ -238,6 +238,17 @@ def sample_input(self) -> dict[str, torch.Tensor]:
...
+    @property
+    def sample_input_rtc(self) -> dict[str, torch.Tensor]:
+        """Return sample input for RTC export.
+
+        ACT does not support RTC, so this returns the same as `sample_input`.
+
+        Returns:
+            A dictionary mapping input names to example torch.Tensor objects.
+        """
+        return self.sample_input
```
Comment (sovrasov) [inline library/src/physicalai/policies/act/model.py]: If ACT can not support RTC, shouldn't this method throw an error instead of returning irrelevant output?

File: library/src/physicalai/policies/smolvla/model.py [hunk 2]
```diff
@@ -327,6 +327,17 @@ def sample_input(self) -> dict[str, torch.Tensor | str]:
...
+    @property
+    def sample_input_rtc(self) -> dict[str, torch.Tensor | str]:
+        """Return sample input for RTC export.
+
+        SmolVLA does not support RTC, so this returns the same as `sample_input`.
+
+        Returns:
+            A dictionary mapping input names to example torch.Tensor objects.
+        """
+        return self.sample_input
```
Comment (sovrasov) [inline library/src/physicalai/policies/smolvla/model.py]: Same as for ACT, the output is not yet relevant

File: library/tests/unit/policies/test_pi05_rtc.py [hunk 3]
```diff
@@ -0,0 +1,760 @@
+# Copyright (C) 2026 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Unit tests for Pi05 Real-Time Chunking (RTC) functionality.
```
Comment (sovrasov) [inline library/tests/unit/policies/test_pi05_rtc.py:4]: Do the tests try to instantiate a full pi05?

File: library/src/physicalai/policies/pi05/model.py [hunk 4]
```diff
@@ -699,6 +699,45 @@ def sample_input(self) -> dict[str, torch.Tensor | str]:
...
+        original_sample_input = self.sample_input
+        rtc_input = {
+            "prev_chunk_left_over": torch.randn(
+                1,
+                chunk_size,
+                max_action_dim,
+                device=device,
+                dtype=torch.float32,
+            ),
+            "inference_delay": torch.tensor(8, device=device, dtype=torch.long),
```
Comment (sovrasov) [inline library/src/physicalai/policies/pi05/model.py]: Do the hardcoded values here always make sense?

File: library/src/physicalai/policies/smolvla/model.py (hunk 2 above)
Comment (eugene123tw) [reply library/src/physicalai/policies/smolvla/model.py]: Removed

File: library/src/physicalai/policies/act/model.py (hunk 1 above)
Comment (eugene123tw) [reply library/src/physicalai/policies/act/model.py]: Removed

File: library/tests/unit/policies/test_pi05_rtc.py (hunk 3 above)
Comment (eugene123tw) [reply library/tests/unit/policies/test_pi05_rtc.py:4]: not the full pi05, only gemma_300m.

```py
model = Pi05Model(
                dataset_stats=dataset_stats,
                paligemma_variant="gemma_300m",
                action_expert_variant="gemma_300m",
                dtype="float32",
                chunk_size=50,
                max_action_dim=32,
                num_inference_steps=2,  # Use few steps for faster tests
            )

```

File: library/src/physicalai/policies/pi05/model.py (hunk 4 above)
Comment (eugene123tw) [reply library/src/physicalai/policies/pi05/model.py]: yeah, it's the number of action steps. So 8 is totally fine. Basically shows how many actions from the beginning of the chunk have already been executed (or are stale) by the time the new prediction arrives.

**Review by sovrasov — APPROVED** [review]
